Glob/Functions/Find-GlobFile.ps1

```diff
@@ -46,84 +59,169 @@ function Find-GlobFile
 
-        [string[]]
         # Any files to exclude. By default, no files are excluded. Any file that gets included that matches an exclude pattern is not returned.
-        $Exclude,
+        [String[]]$Exclude,
 
-        [Switch]
         # By default, the search is case-insensitive. To perform a case-sensitive search, use this switch.
-        $CaseSensitive
+        [switch]$CaseSensitive,
+
+        # Include hidden files, too.
+        [switch]$Force
     )
 
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
     
-    foreach( $rootPath in $Path )
+    function Test-GlobMatch
     {
-        $rootPath = Resolve-Path -Path $rootPath | Select-Object -ExpandProperty 'ProviderPath'
-        if( -not $rootPath )
-        {
-            continue
-        }
-
-        $options = New-Object 'DotNet.Globbing.GlobOptions'
-        $options.Evaluation.CaseInsensitive = -not $CaseSensitive
+        param(
+            [object]$InputObject,
 
-        $includeGlobs = $Include | Where-Object { $_ } | ForEach-Object { [DotNet.Globbing.Glob]::Parse($_,$options) }
-        $excludeGlobs = $Exclude | Where-Object { $_ } | ForEach-Object { [DotNet.Globbing.Glob]::Parse($_,$options) }
+            [switch]$IsDirectory
+        )
 
-        Push-Location -Path $rootPath
+        $relativePath = Resolve-Path -LiteralPath $InputObject.FullName -Relative 
+        if( $relativePath.Length -ge 2 -and `
+            $relativePath[0] -eq '.' -and `
+            ($relativePath[1] -eq [IO.Path]::DirectorySeparatorChar -or `
+            $relativePath[1] -eq [IO.Path]::AltDirectorySeparatorChar))
+        {
+            # Remove the .\ or ./ at the beginning of the path, as the glob library doesn't like it.
+            $relativePath = $relativePath.Substring(2)
+        }
+        
+        $result = ' '
+        $whatMatched = ''
+        $showMessage = -not $IsDirectory
         try
         {
-            foreach( $fileInfo in (Get-ChildItem -Path $rootPath -File -Recurse) )
+            if( -not $IsDirectory )
             {
-                $relativePath = 
-                    $fileInfo.FullName | 
-                    Resolve-Path -Relative | 
-                    ForEach-Object { $_ -replace '^\.(\\|/)','' }
                 $matches = $false
                 foreach( $includeGlob in $includeGlobs )
                 {
                     if( $includeGlob.IsMatch($relativePath) )
                     {
+                        $whatMatched = $includeGlob
                         $matches = $true
                         break
                     }
                 }
 
                 if( -not $matches )
                 {
-                    continue
+                    return $false
                 }
+            }
 
-                foreach( $excludeGlob in $excludeGlobs )
+            $matches = $true

```

Comment:
`$Matches` is an [automatic variable](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-5.1#matches) related to the `-match` operator.

Use a different name for this variable?